### PR TITLE
Warn when connection to Spamhaus times out

### LINKED ISSUE #1816: Spamhaus timeout leads to false positive
>> https://discourse.mailinabox.email/t/spamhaus-time-out-notification/6879

A user got a message that his/her mailserver was blacklisted. I think it was only a timeout and that we should

* check for [timeout] as return code and set another error message then ("We could not verify if your mail server is blacklisted, since the connection to Spamhaus timed out.") or

* check for specific return codes from spamhaus to verify that we are blacklisted

### PR DESCRIPTION
**Pull request by fspoettel**
This prints a warning rather than an error when a `Timeout` is raised and `query_dns()` then returns `"[timeout]"`.
Decided against explicit checks for Spamhaus error codes ([zen](https://www.spamhaus.org/zen/) [dbl](https://www.spamhaus.org/faq/section/Spamhaus%20DBL#291)) for now to keep it simple.

closes #1816

### CODE REVIEW

File: management/status_checks.py [hunk 1]
```diff
@@ -293,6 +293,8 @@ def run_network_checks(env, output):
 	zen = query_dns(rev_ip4+'.zen.spamhaus.org', 'A', nxdomain=None)
 	if zen is None:
 		output.print_ok("IP address is not blacklisted by zen.spamhaus.org.")
+	elif zen == "[timeout]":
+		output.print_warning("Connection to zen.spamhaus.org timed out. We could not determine whether your server's IP address {} is blacklisted. Please try again later.".format(rev_ip4))
```
Comment (jvolkenant) [inline management/status_checks.py:297]: I don't think we want to display to the user their reversed IP; would cause confusion

File: management/status_checks.py [hunk 2]
```diff
@@ -675,9 +677,11 @@ def check_mail_domain(domain, env, output):
 	# Stop if the domain is listed in the Spamhaus Domain Block List.
 	# The user might have chosen a domain that was previously in use by a spammer
 	# and will not be able to reliably send mail.
-	dbl = query_dns(domain+'.dbl.spamhaus.org', "A", nxdomain=None)
+	dbl = query_dns(domain+".dbl.spamhaus.org", "A", nxdomain=None)
```
Comment (jvolkenant) [inline management/status_checks.py:680]: Why the change from single to double quotes?

Comment (fspoettel) [reply management/status_checks.py:680]: the rest of the file is using them, I see I missed the occurrence in 293

File: management/status_checks.py (hunk 1 above)
Comment (fspoettel) [reply management/status_checks.py:297]: thanks, fair point.

File: management/status_checks.py [hunk 3]
```diff
@@ -293,6 +293,8 @@ def run_network_checks(env, output):
 	zen = query_dns(rev_ip4+'.zen.spamhaus.org', 'A', nxdomain=None)
 	if zen is None:
 		output.print_ok("IP address is not blacklisted by zen.spamhaus.org.")
+	elif zen == "[timeout]":
+		output.print_warning("Connection to zen.spamhaus.org timed out. We could not determine whether your server's IP address is blacklisted. Please try again later.")
```
Comment (ddavness) [inline management/status_checks.py:297]: Maybe we could add a the website endpoint that we point the users to when the IP/domain turns out to be legitimately blocked (`http://www.spamhaus.org/query/ip/<ipv4>`)

File: management/status_checks.py [hunk 4]
```diff
@@ -678,6 +680,8 @@ def check_mail_domain(domain, env, output):
 	dbl = query_dns(domain+'.dbl.spamhaus.org', "A", nxdomain=None)
 	if dbl is None:
 		output.print_ok("Domain is not blacklisted by dbl.spamhaus.org.")
+	elif dbl == "[timeout]":
+		output.print_warning("Connection to dbl.spamhaus.org timed out. We could not determine whether the domain {} is blacklisted. Please try again later.".format(domain))
```
Comment (ddavness) [inline management/status_checks.py:684]: Same here: `http://www.spamhaus.org/query/domain/example.com`

**Review by ddavness — COMMENTED** [review]
Bumping this PR as I have a couple suggestions :^)

**Comment by JoshData** [comment]
This issue has never happened for me and I assume it is intermittent, so we don't need to make the error message more informative. Merging!

**Comment by ddavness** [comment]
I think it may depend on who your providers are - for [Hetzner](https://hetzner.com/), for whatever reason all Spamhaus queries take either very long (in the ballpark of the 30 seconds) or just time out.
